test/e2e/e2e_local_source_upload_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/test/utils"
+)
+
+var _ = Describe("For a Kubernetes cluster with Tekton and build installed", func() {
+	var (
+		testID   string
+		build    *buildv1alpha1.Build
+		buildRun *buildv1alpha1.BuildRun
+	)
+
+	AfterEach(func() {
+		if buildRun != nil {
+			testBuild.DeleteBR(buildRun.Name)
+			buildRun = nil
+		}
+
+		if build != nil {
+			testBuild.DeleteBuild(build.Name)
+			build = nil
+		}
+	})
+
+	Context("when LocalCopy BuildSource is defined", func() {
+		BeforeEach(func() {
+			testID = generateTestID("local-copy")
+			build = createBuild(
+				testBuild,
+				testID,
+				"test/data/build_buildah_cr_local_source_upload.yaml",
+			)
+		})
+
+		It("should generate LocalCopy TaskRun, using the waiter", func() {
+			var err error
+			buildRun, err = buildRunTestData(
+				testBuild.Namespace,
+				testID,
+				"test/data/buildrun_buildah_cr_local_source_upload.yaml",
+			)
+			Expect(err).ToNot(HaveOccurred(), "Error retrieving buildrun test data")
+
+			validateWaiterBuildRun(testBuild, buildRun)
+		})
+	})
+})
+
+func getBuildRunStatusCondition(name types.NamespacedName) *buildv1alpha1.Condition {
+	testBuildRun, err := testBuild.LookupBuildRun(name)
+	Expect(err).ToNot(HaveOccurred(), "Error retrieving the BuildRun")
+
+	if len(testBuildRun.Status.Conditions) == 0 {
+		return nil
+	}
+	return testBuildRun.Status.GetCondition(buildv1alpha1.Succeeded)
+}
+
+// validateWaiterBuildRun assert the BuildRun informed will fail, since Waiter's timeout is reached
+// and it causes the actual build process to fail as well.
+func validateWaiterBuildRun(testBuild *utils.TestBuild, testBuildRun *buildv1alpha1.BuildRun) {
+	// the build run should be created only once in the cluster, so if the GetBR function fails then
+	// it proceeds to create the new instance asserting its success
+	if _, err := testBuild.GetBR(testBuildRun.Name); err != nil {
+		Expect(testBuild.CreateBR(testBuildRun)).
+			ToNot(HaveOccurred(), "Failed to create BuildRun")
+	}
```

Comment:
What would actually be the scenario where the BuildRun exists already? Because of `generateTestID`, the BuildRun name will be unique and you have cleanup as in the other tests.

Reply:
In fact, that was a left over, so I'm tweaking this part to become a regular `BuildRun` creation. Please consider.